src/libutil/git.hh

```diff
@@ -5,8 +5,160 @@
 #include <string_view>
 #include <optional>
 
+#include "types.hh"
+#include "serialise.hh"
+#include "hash.hh"
+#include "source-accessor.hh"
+#include "fs-sink.hh"
+
 namespace nix::git {
 
+enum struct ObjectType {
+    Blob,
+    Tree,
+    //Commit,
+    //Tag,
+};
+
+using RawMode = uint32_t;
+
+enum struct Mode : RawMode {
```

Comment:
Can we use the libgit2 types here? https://github.com/libgit2/libgit2/blob/HEAD/include/git2/types.h#L222-229

And in general it would probably be nicer to use libgit2 instead of having our own Git dump/parse functions.